sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -800,6 +804,102 @@ sycl::kernel k_float = kb.ext_oneapi_get_kernel("bartmpl<float>");
 sycl::kernel k_int = kb.ext_oneapi_get_kernel("bartmpl<int>");
 ----
 
+=== New kernel bundle member functions for device globals
+
+This extensions adds the following new `kernel_bundle` member functions to let
+the host application interact with device globals defined in runtime-compiled
+code. Device globals are only supported for the `source_language::sycl`
+language.
+
+[source,c++]
+----
+namespace sycl {
+
+template <bundle_state State>
+class kernel_bundle {
+  // Continued from "New kernel bundle member functions"
+
+  bool ext_oneapi_has_device_global(const std::string &name);
+  void *ext_oneapi_get_device_global_address(const std::string &name,
+                                             const device &dev);
+  size_t ext_oneapi_get_device_global_size(const std::string &name);
+};
+
+} // namespace sycl
+----
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+bool ext_oneapi_has_device_global(const std::string &name)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Returns:_ The value `true` only if
+
+* the kernel bundle was created from a bundle of state
+  `bundle_state::ext_oneapi_source` in the language `source_language::sycl`, and
+* it defines a device global whose name is `name`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+void *ext_oneapi_get_device_global_address(const std::string &name,
+                                           const device &dev)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Returns:_ A device USM pointer to the storage for the device global `name` on
+device `dev`.
+
+_Remarks:_ The contents of the device global may be read or written from the
+host by reading from or writing to this address. If the address is read before
+any kernel writes to the device global, the read operation returns the device
+global's initial value.
```

Comment:
What is the intended usage of these functions?

`sycl_ext_oneapi_device_global` defines explicit `copy` overloads that take a `device_global`, and as far as I can tell there was no way before to get the address and size of a `device_global` from the host.

I might have misunderstood something, but the current situation feels weird and incomplete to me. It seems like a developer working with a `device_global` would need to know whether the global is defined in their regular SYCL program or via run-time compiled source, and then use different functions to read/write the global.  Is that right?  If so, why can't we use the same mechanism everywhere?

Also tagging @steffenlarsen to look at this, because he knows a lot more about `device_global` than I do.

Reply:
The design of the API was done by @gmlueck as part of the module API and feedback he gave on an earlier PR.
AFAIK, the API was designed to meet customer requirements and align with existing APIs, but @gmlueck might have more background on this.